Ab issue 13

AllisonBC opened the pull request:
## Summary
Closes Issue \(https://github.com/dansjack/covid-score-app/issues/13) and probably Issue 12

Implements fragment-based architecture for app with a FragmentPagerAdapter, UI to go from screen to screen, skeleton UI elements for future implementation as well as a form for user to enter a location manually. 

If there are many changes, make a list of them
* FragmentAdapter.java
* GpsPageFragment.java
* LocationFindOptionsFragment.java
*LocationSelectionPageFragment.java
*RiskDetailPageFragment.java
*WelcomePageFragment.java
*fragment_gps_page.xml
*fragment_location_find_options.xml
*fragment_location_selection.xml
*fragment_risk_detail.xml
*fragment_welcome_page.xml

****Known bug: fragments aren't stable for UI like 'Back' etc

## Checklist
- [ ] New tests added to account for changes in this MR
- [x] All tests passing locally
- [x] All tests passing on CircleCI
- [x] Assigned at least 2 reviewers

AllisonBC committed:
Add bare bones form layout

AllisonBC committed:
Move form to fragment; spinner not functioning properly

AllisonBC committed:
Structure and style changes; spinners not functioning properly

AllisonBC committed:
Add layouts and fragment

Add LocationSelectionFragment for location spinner screen

AllisonBC committed:
Implement fragment-based architecture

AllisonBC committed:
Fixed buttons

dansjack reviewed (commented):
The app didn't build properly, but I think you can fix it by removing the `loadsView()` test inside `MainActivityTest.java` as well as the `@RunWith` decorator in the same file
<img width="700" alt="image" src="https://user-images.githubusercontent.com/30481844/96521108-59865d80-1225-11eb-8752-bfafcaa0aecd.png">

I pointed the PR from merging into `main` to merging into `work/room-api-ui` since that's where the room and api code is right now (you'll probably have a few merge conflicts now). Once we get more tests and have everything a bit more glued together we can merge into `main`

AllisonBC committed:
Fixes for merge

Delete erroneous test code; fix fragment_risk_detail.xml so future additions to file won't cause ScrollView children error

AllisonBC commented:
Removed the loadsView() test as well as the @RunWith decorator inside MainActivityTest.java. Thanks @dansjack !

AllisonBC committed:
UI updates

Remove GPS button and move location search to front screen

AllisonBC committed:
Merge remote-tracking branch 'origin/work/room-api-ui' into ab-issue-13

AllisonBC commented:
Merged room-api-ui branch into [issue 13](https://github.com/dansjack/covid-score-app/tree/ab-issue-13) branch

AllisonBC committed:
Fix merge errors

dansjack reviewed (commented):
The app crashed with this error when I tried to boot it up 
<img width="700" alt="image" src="https://user-images.githubusercontent.com/30481844/96940560-df541400-1484-11eb-8370-173a7eaf3b5b.png">

AllisonBC committed:
Fix crashing bug in Main

Bug fix in Main and minor build.gradle updates

AllisonBC committed:
gradle reversions

dansjack committed:
fix jacoco

unicornsleuth commented on app/src/androidTest/java/com/nsc/covidscore/MainActivityTest.java:
```diff
@@ -1,38 +1,13 @@
     public ActivityTestRule<MainActivity> activityTestRule
             = new ActivityTestRule<>(MainActivity.class, true, true);
 
-    @Test
```
I'm confused as to why so much of this file was deleted. Was it just because the loadsViews() failed? Why did so many of the imports disappear? And the @RunWith?

dansjack replied:
There was a build error caused by parts of MainActivityTest.java https://github.com/dansjack/covid-score-app/pull/24#pullrequestreview-512224176

unicornsleuth commented on app/src/main/java/com/nsc/covidscore/FragmentAdapter.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public void addFragment(Fragment fragment, String title){
+        mFragmentList.add(fragment);
```
Do we need to do any checks to make sure it's unique & we don't get multiple copies of the same fragment? I know we probably won't make that mistake, but maybe it's a good idea to add that. Or possibly you could make it a Set so the uniqueness is enforced? Just a thought.

AllisonBC replied:
The setupView Pager method uses the addFragment method to add the fragments to an adapter just after the adapter is instantiated. Since fragments are created with getItem via the fragment's specific index in the list, I don't think having an extra fragment in the list, even if it somehow happened, would be an issue.

AllisonBC replied:
Resolving for now. We can create a bug issue for this if I misinterpreted you here, but it shouldn't need to hold up the merge

unicornsleuth commented on app/src/main/java/com/nsc/covidscore/GpsPageFragment.java:
```diff
@@ -0,0 +1,91 @@
+
+import androidx.fragment.app.Fragment;
+
+/**
```
In general, we want to avoid (as much as possible) merging in commented-out code. But I understand this part is a WIP : )

AllisonBC replied:
I kept it because I thought we might need it if we added functionality for user to later decide to determine location via GPS, but even then we won't need a screen for that. I'll delete

unicornsleuth commented on app/src/main/java/com/nsc/covidscore/LocationFindOptionsFragment.java:
```diff
@@ -0,0 +1,94 @@
+//package com.nsc.covidscore;
```
Since this is all commented out, can you actually remove this from the PR? Like, c/p the file, delete it, push it, then you can add it back in whatever branch you're working on. Actually maybe you could do that with the GpsPageFragment as well, since it's not being called/used currently (I think).

unicornsleuth commented on app/src/main/java/com/nsc/covidscore/LocationManualSelectionFragment.java:
```diff
@@ -0,0 +1,343 @@
+        // Required empty public constructor
+    }
+
+//TODO: this is all boilerplate code for doing a more graceful transition bet fragments; delete if we don't want
```
This is fine for now, but in our next meeting we can talk about this & make a decision on whether we want to implement it or not, so that we don't just leave a ton of commented-out code here indefinitely? From my perspective, I don't know why we wouldn't want more graceful transitions between fragments, but that can be a separate issue. : )

AllisonBC replied:
I've deleted the unused code for now. It's easily reproduceable if we want it later

unicornsleuth commented on app/src/main/java/com/nsc/covidscore/MainActivity.java:
```diff
@@ -44,18 +48,29 @@
 
         // Access to Room Database
-        vm = new ViewModelProvider(this).get(CovidSnapshotWithLocationViewModel.class);
+        vm = new ViewModelProvider(this).get(CovidSnapshotWithLocationViewModel .class);
```
Is this space intentional? I don't get why the view model works at all with the space in there... But I tested & it does... idk

AllisonBC replied:
How weird AS didn't call that out on my end, in any way. I'll fix it

unicornsleuth commented on app/src/main/res/layout/covid_detail_card.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Where is this being used? Was this intended to be the card for displaying the COVID stats?

AllisonBC replied:
It's not used right now. I added it thinking I was doing all UI, and then looked over the description of my issue again and saw I was exceeding the bounds of the issue's requirements. I'll delete so you can implement the cards how you'd like.

AllisonBC committed:
Delete obsolete code

AllisonBC committed:
Delete unused code and add androidx multidex

Delete unused strings and commented out code; add androidx multidex to app level build.gradle to fix error

AllisonBC committed:
Remove unused files; fix typo

dansjack reviewed (approved):
app no longer crashes

AllisonBC committed:
Merge branch 'work/room-api-ui' into ab-issue-13

AllisonBC committed:
Remove outdated failing test from MainActivityTest
